ably/rest_client.go

```diff
@@ -470,12 +488,29 @@ func (p *HTTPPaginatedResponse) HasNext(ctx context.Context) bool {
 	return p.nextLink != ""
 }
 
-// Items unmarshals the current page of results as JSON into the provided
+// Items unmarshals the current page of results into the provided
 // variable.
 //
 // See the "Paginated results" section in the package-level documentation.
 func (p *HTTPPaginatedResponse) Items(dst interface{}) error {
-	return json.Unmarshal(p.items, dst)
+	typ, _, err := mime.ParseMediaType(p.PaginatedResult.res.Header.Get("Content-Type"))
+	if err != nil {
+		return err
+	}
+
+	/// Turn non arrays into arrays so it fits the output kind
+	if typ == "application/json" {
+		token, _ := json.NewDecoder(bytes.NewReader(p.items)).Token()
+		if token != json.Delim('[') {
+			p.items = append(Raw{'['}, p.items...)
+			p.items = append(p.items, ']')
+		}
+	} else if typ == "application/x-msgpack" {
+		if (p.items[0]&0xf0 != 0x90) && p.items[0] != 0xdc && p.items[0] != 0xdd {
+			p.items = append(Raw{0x91}, p.items...)
```

Comment:
I'm having a hard time reading lines 509 and 510. Maybe some commenting is needed here?

`0x91` looks like a single left quote ( https://bytetool.web.app/en/ascii/code/0x91/ ) I'm not sure how appending this character converts the raw bytes to an array. Please can you talk me through these lines.

Reply:
messagepack is a binary format. The nibble 9 is used for the array type, followed by the length. So 91 is an array of length one.